docs/preview/logdecorations.md

```diff
@@ -0,0 +1,238 @@
+
+# Log decorations
+
+
+Task authors should be able to control how the log output is displayed to the end user.
+This outlines different decoration options that are available.
+
+## Collapse
+
+>This is a draft spec and is subject to change
+
+>Note that that if you log an error using ```##vso[task.logissue]error/warning message``` command (see [logging commands](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md) here) we will surface those errors in build view and when clicked , we will automatically jump to that particular line. If it's already part of a group, we will auto-expand the group.
+
+Task authors can mark any part of the log as a collapsible region using these decorations:
+
+Starting the collapsible region - `##[group:${optionalGroupName}:start]`
+
+Ending the collapsible region - `##[group:${optionalGroupName}:end]`
+
+`${optionalGroupName}` is optional.
+
+Note: 
+    
+*  The first line of region will be taken as group title by default.
+*  If there's only one line in the region (including the group title), it will not be considered as a collapsible
+*  If there's `##[group:start]` with out corresponding `##[group:end]` we will add implicit `##[group:end]`
```

Comment:
Do we want to support nested group? if so, how do we implicit add `##[group:end]`?

Reply:
nested groups will be possible if author sends the groupname that's optional. Example covers this aspect.
So in that case, we will end that group only at the end of parsing.
If someone adds `##[group:group1:start]` and no end at all, we will consider the whole text after that as part of that group. Implict `##[group:end]` only applies to non-named groups.
I will add this information to the doc.